### ISSUE TITLE
different indentation for function arguments of same nesting level expected

### ISSUE THREAD

**Issue by colinjoy** [eslint/eslint#6890]
Version: `1.10.3`
Parser: `default`
Rule: `"indent": 2`

``` javascript
// EX.1 eslint complains
lib.then(function() {
    lib.waitFor(function then() {
            x = 1;
        }, // A
        function onTimeout() {
            x = 2;
        }, // B
        500
    );
});

// Ex.2 eslint happy
lib.then(function() {
    lib.waitFor(function then() {
        x = 1;
    }, // A
        function onTimeout() {
            x = 2;
        }, // B
        500
    );
});
```

this does not appear to be right.
A and B are on the same nesting level, yet ESLint approves EX.2 instead of EX.1

**Comment by platinumazure** [comment]
I didn't realize you were on ESLint 1.x. Can you please try with latest to see if this still happens? Thanks!

**Comment by colinjoy** [comment]
the same appears to be true using eslint 3.3.0

```
// Ex.1 eslint unhappy
lib.then(function() {
    lib.waitFor(function then() {
            x = 1;
        }, // A
        function onTimeout() {
            x = 2;
        }, // B
        500
    );
});

// Ex.2 eslint happy
lib.then(function() {
    lib.waitFor(function then() {
        x = 1;
    }, // A
    function onTimeout() {
        x = 2;
    }, // B
    500
    );
});

// Ex.3 eslint happy
lib.then(function() {
    lib.waitFor(function then() {
        x = 1;
    }, // A
        function onTimeout() {
            x = 2;
        }, // B
        500
    );
});
```

**Comment by gyandeeps** [comment]
It works correctly if you just have 1 argument for a fucntion then you can inline the first argument on the same line has opening brace like this

```js
lib.then(function() {
    lib.waitFor(function then() {
        x = 1;
    });
});
```

But in your case you have more than one argument, so it expects something like this

```js
lib.then(function() {
    lib.waitFor(
        function then() {
            x = 1;
        }, // A
        function onTimeout() {
            x = 2;
        }, // B
        500
    );
});
```

**Comment by colinjoy** [comment]
we were able to adapt our code-style so it pleases eslint.
thank you for the input.

**Comment by isonmad** [comment]
This still potentially affects other people and nothing in eslint has changed, couldn't this still be open?
